### PR TITLE
Added Empty space, orbit controls, space hover outline, and background grid

### PR DESCRIPTION
**Pull request by JamesCoyne**
To my knowledge, we don't have any open issues pertaining to this, but I did it anyway!

**The long and short of these changes:**
- Maze Info now has a boolean variable called 'path' to store if the space is the selected path or not
- Orbit controls have been added to the canvas
- There is also a background grid in the canvas
- Generic Space has been split into two sub-components, wall and empty, where wall renders a cube with a texture, and empty renders a colored plane. This is done to give both of them different meshes, materials, and textures. 
- Empty spaces have been added! they have two boolean props,  `visited` and `path` and currently render as different colors pertaining to this.

**Further work and things to consider**
- Personally I think we might consider a better way to store `path` and `visited` in MazeInfo, for one thing, currently walls are given these props while only empty spaces will ever be visited/used as a path.

- It might be good to eventually split `startPoint` and `endPoint` into their own subcomponents as well just for readability, as well as if they end up having their own custom mesh or material

- three.js's raycasting system has an issue where it doesn't occlude objects, meaning that sometimes if the camera is angled correctly you can select like two or three spaces at once. I'm guessing that there might be an easy fix for this. The problem isn't mission-critical, but it does detract from the user experience and it's worth investigating further as a stretch goal.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "skipLibCheck": true,
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true,
-    "strict": true,
+    "strict": false,
```
Comment (uptonm) [inline tsconfig.json]: Just curious, why false?

**Comment by JamesCoyne** [comment]
Oop, good catch on the .code-workspace. I turned off strict mode to try suppress some issues I was having, but it’s not essential and should be set to true. I’ll fix and recommit shortly.

**Comment by uptonm** [comment]
Yea, no worries. If there are still issues feel free to leave it as false. I was just curious :) good work

**Comment by JamesCoyne** [comment]
Thanks! Just fixed it, ready to merge if y'all see no other issues :^)

**Pull request merged**